src/telemetry_subscriber.rs

```diff
@@ -130,50 +153,43 @@ impl TelemetrySubscriber {
 }
 
 impl Subscriber for TelemetrySubscriber {
-    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
-        metadata.level() <= &self.log_level
+    fn enabled(&self, _metadata: &Metadata<'_>) -> bool {
+        true
     }
 
     fn new_span(&self, span: &Attributes<'_>) -> Id {
         let new_span = self.build_span(span);
-        let id = loop {
-            // collision is extremely unlikely (random u64), handle anyway
-            // v. unlikely but possible: identical id is gen'd between 'contains_key' and 'insert'
-            let id = generate_span_id();
-            if !self.spans.contains_key(&id) {
-                break id;
-            };
-        };
 
-        self.spans.insert(
-            id.clone(),
-            RefCt {
+        let idx: usize = self
+            .spans
+            .insert(RwLock::new(RefCt {
                 ref_ct: 1,
                 inner: new_span,
-            },
-        );
+            }))
+            .expect("unable to add span to slab (OOM?)");
 
-        id
+        idx_to_id(idx)
     }
 
     // record additional values on span map
-    fn record(&self, span: &Id, values: &Record<'_>) {
-        if let Some(mut span_data) = self.spans.get_mut(&span) {
+    fn record(&self, id: &Id, values: &Record<'_>) {
+        if let Some(rw_lock) = self.spans.get(id_to_idx(id)) {
+            let mut span_data = rw_lock.write().unwrap();
             let mut visitor = HoneycombVisitor {
                 accumulator: &mut span_data.values,
             };
             values.record(&mut visitor);
         } else {
             println!(
                 "no span in map when recording to span with id {:?}, possible bug",
-                span
+                id
             )
         }
     }
 
     fn record_follows_from(&self, _span: &Id, _follows: &Id) {}
 
-    // record event (publish directly to telemetry, not a span)
+    // record event (pub(crate)lish directly to telemetry, not a span)
```

Comment:
lol, looks like an over-eager find and replace:
```suggestion
    // record event (publish directly to telemetry, not a span)
```